modules/top-level/test.nix

```diff
@@ -19,23 +90,79 @@ in
       description = "The test derivation's name.";
     };
 
+    buildNixvim = lib.mkOption {
+      type = lib.types.bool;
+      description = "Whether to build the nixvim config in the test.";
+      default = true;
+    };
+
     runNvim = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to run `nvim` in the test.";
-      default = true;
+      defaultText = lib.literalExpression "config.test.buildNixvim";
+      default = cfg.buildNixvim;
     };
 
     checkWarnings = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to check `config.warnings` in the test.";
+      apply = x: lib.warnIfNot x "`test.checkWarnings = false` is replaced with `test.warnings = [ ]`." x;
       default = true;
     };
 
     checkAssertions = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to check `config.assertions` in the test.";
+      apply =
+        x: lib.warnIfNot x "`test.checkAssertions = false` is replaced with `test.assertions = [ ]`." x;
       default = true;
     };
+
+    warnings = lib.mkOption {
+      type = lib.types.listOf expectationType;
+      description = "A list of expectations for `warnings`.";
+      defaultText = lib.literalMD "Expect `count == 0`";
+      default = [ 0 ];
+    };
+
+    assertions = lib.mkOption {
+      type = lib.types.listOf expectationType;
+      description = "A list of expectations for `assertions`.";
+      defaultText = lib.literalMD "Expect `count == 0`";
+      default = [ 0 ];
+    };
+
+    predefinedExpectationPredicates = lib.mkOption {
```

Comment:
I think `predefined` is a bit of mis-nomer, a test could add more predicates as the option is not read-only. Maybe something like `named` or `registered`?